tests/migrations/test_autodetector.py

```diff
@@ -2361,6 +2361,32 @@ def test_multiple_bases(self):
         self.assertOperationAttributes(changes, "app", 0, 3, name="D")
         self.assertOperationAttributes(changes, "app", 0, 4, name="E")
 
+    def test_metaclass(self):
+        class AlphaBase(type):
+            pass
+
+        class Alpha(metaclass=AlphaBase):
+            pass
+
+        Success = ModelState(
+            "testapp",
+            "Success",
+            [("a_id", models.AutoField(primary_key=True))],
+            bases=(Alpha, models.Model),
+        )
```

Comment:
This is an invalid model

```
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

In other words `AlphaBase` must subclass `ModelBase` to be a valid model.

This makes me believe that this test doesn't cover the adjusted `render` code.

Reply:
@charettes where are you seeing this error?

Reply:
@ethanhowell try defining a model matching this state

```python
from django.db import models

class AlphaBase(type):
    pass

class Alpha(metaclass=models.base.ModelBase):
    pass

class Foo(Alpha, models.Model):
    pass
```

Reply:
What I'm showing here is that the ModelState doesn't really need the metaclass because it will be generated. Sure that model would fail if written out like ^ but the migration would still work

Reply:
My point is that you could not define a model to create this migration from it in the first place so the issue reported in the ticket has to be subtly different from what you're testing here. There's no point in testing an impossible to reach migration state.

Looks like this previous attempt at a test was more appropriate https://github.com/django/django/pull/5907/files#diff-805fc0a0279c554968f653cd678b0632R912-R941

Reply:
@charettes ok I know see what you're talking about, thanks. I changed the test to generate the model state from a real model then try and render that. What do you think?